url/urlencoded-parser.html

```diff
@@ -0,0 +1,71 @@
+<!doctype html>
+<meta charset=utf-8>
+<title>application/x-www-form-urlencoded parser test</title>
+<script src=/resources/testharness.js></script>
+<script src=/resources/testharnessreport.js></script>
+<div id=log></div>
+<script>
+[
+  { "input": "test", "output": [["test", ""]] },
+  { "input": "\uFEFFtest=\uFEFF", "output": [["\uFEFFtest", "\uFEFF"]] },
+  { "input": "%EF%BB%BFtest=%EF%BB%BF", "output": [["\uFEFFtest", "\uFEFF"]] },
+  { "input": "%FE%FF", "output": [["\uFFFD\uFFFD", ""]] },
+  { "input": "%FF%FE", "output": [["\uFFFD\uFFFD", ""]] },
+  { "input": "†&†=x", "output": [["†", ""], ["†", "x"]] },
+  { "input": "%C2", "output": [["\uFFFD", ""]] },
+  { "input": "%C2x", "output": [["\uFFFDx", ""]] },
+  { "input": "_charset_=windows-1252&test=%C2x", "output": [["_charset_", "windows-1252"], ["test", "\uFFFDx"]] },
+  { "input": '', "output": [] },
+  { "input": 'a', "output": [['a', '']] },
+  { "input": 'a=b', "output": [['a', 'b']] },
+  { "input": 'a=', "output": [['a', '']] },
+  { "input": '=b', "output": [['', 'b']] },
+  { "input": '&', "output": [] },
+  { "input": '&a', "output": [['a', '']] },
+  { "input": 'a&', "output": [['a', '']] },
+  { "input": 'a&a', "output": [['a', ''], ['a', '']] },
+  { "input": 'a&b&c', "output": [['a', ''], ['b', ''], ['c', '']] },
+  { "input": 'a=b&c=d', "output": [['a', 'b'], ['c', 'd']] },
+  { "input": 'a=b&c=d&', "output": [['a', 'b'], ['c', 'd']] },
+  { "input": '&&&a=b&&&&c=d&', "output": [['a', 'b'], ['c', 'd']] },
+  { "input": 'a=a&a=b&a=c', "output": [['a', 'a'], ['a', 'b'], ['a', 'c']] },
+  { "input": 'a==a', "output": [['a', '=a']] },
+  { "input": 'a=a+b+c+d', "output": [['a', 'a b c d']] },
+  { "input": '%=a', "output": [['%', 'a']] },
+  { "input": '%a=a', "output": [['%a', 'a']] },
+  { "input": '%a_=a', "output": [['%a_', 'a']] },
+  { "input": '%61=a', "output": [['a', 'a']] },
+  { "input": '%61+%4d%4D=', "output": [['a MM', '']] }
```

Comment:
I haven't done that so far. I guess I'll stick with my own convention, but if you ever change this feel free to add it.